src/inmanta/ast/entity.py

```diff
@@ -274,39 +288,6 @@ def get_attribute(self, name: str) -> Optional["Attribute"]:
                     return attr
         return None
 
-    @memoize
-    def __get_related(self) -> "Set[Entity]":
-        # down
```

Comment:
In the new check we don't seem to go down anymore. Then again, I see no reason at all why we would want to do so. It was a conscious decision to drop this?

Reply:
The reason we had to go up and down is that the tree is still being built up.

If the tree is complete, and we check for every relation, we only have to check up. Because then the lower one will find the higher one. 

If the tree is incomplete, we have to check both up and down, because if we are the upper one and we have been added last, we can not just check up. 

(Or am I wrong?)

Reply:
I see. I think you're right.